Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -68,7 +76,17 @@ def ensure_running(self):
                 exe = spawn.get_executable()
                 args = [exe] + util._args_from_interpreter_flags()
                 args += ['-c', cmd % r]
+                # Register a signal mask that will block the signals. This signal
+                # mask will be inherited by the child that is going to be spawned
+                # and will protect the child from a race condition that can make
+                # the child die before it registers signal handlers for SIGINT and
+                # SIGTERM. The mask is unregistered after spawning the child.
+                # For more info, see: https://bugs.python.org/issue33613
+                if _HAVE_SIGMASK:
+                    signal.pthread_sigmask(signal.SIG_BLOCK, _IGNORED_SIGNALS)
                 pid = util.spawnv_passfds(exe, args, fds_to_pass)
+                if _HAVE_SIGMASK:
```

Comment:
This should use a `try..finally` block.

Reply:
Oops, right. That's my fault, I proposed the code to @pablogsal :-)